src/test/feature/ManagementTool/test_hawq_register.cpp

```diff
@@ -3,18 +3,36 @@
 #include "lib/command.h"
 #include "lib/sql_util.h"
 #include "lib/string_util.h"
+#include "lib/hdfs_config.h"
 
 #include "gtest/gtest.h"
 
 using std::string;
 using hawq::test::SQLUtility;
 using hawq::test::Command;
+using hawq::test::HdfsConfig;
 
 /* This test suite may consume more than 80 seconds. */
 class TestHawqRegister : public ::testing::Test {
  public:
   TestHawqRegister() {}
   ~TestHawqRegister() {}
+  string getHdfsLocation() {
+    HdfsConfig hc;
+    string hostname = "";
+    int port = 0;
+    if (hc.isHA()) {
+      hc.getActiveNamenode(hostname, port);
```

Comment:
Will this work? Shouldn't pass the pointer? If the params are modified inside function getActiveNamenode, I don't think they will be modified after the function exists. Your thought?